types/blob/blob.go

```diff
@@ -0,0 +1,67 @@
+package blob
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/rollkit/celestia-openrpc/types/namespace"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// Commitment is a Merkle Root of the subtree built from shares of the Blob.
+// It is computed by splitting the blob into shares and building the Merkle subtree to be included
+// after Submit.
+type Commitment []byte
+
+func (com Commitment) String() string {
+	return string(com)
+}
+
+// Equal ensures that commitments are the same
+func (com Commitment) Equal(c Commitment) bool {
+	return bytes.Equal(com, c)
+}
+
+// Blob represents any application-specific binary data that anyone can submit to Celestia.
+type Blob struct {
+	tmproto.Blob `json:"blob"`
+
+	Commitment Commitment `json:"commitment"`
+}
+
+// Namespace returns blob's namespace.
+func (b *Blob) Namespace() namespace.ID {
+	return append([]byte{uint8(b.NamespaceVersion)}, b.NamespaceId...)
+}
+
+type jsonBlob struct {
+	Namespace    namespace.ID `json:"namespace"`
+	Data         []byte       `json:"data"`
+	ShareVersion uint32       `json:"share_version"`
+	Commitment   Commitment   `json:"commitment"`
+}
+
+func (b *Blob) MarshalJSON() ([]byte, error) {
```

Comment:
We should have unit testing around marshalling and unmarshalling

Reply:
The `Blob` type (along with other types) are temporarily copied here until dependency issue with `celestia-node` is resolved. Also the API has been changing and might change until release.

There are tests in the source where this was copied from: https://github.com/celestiaorg/celestia-node/blob/main/blob/blob_test.go

Reply:
Deferring until API is stable to copy test cases for `Blob` (along with the rest of the types as none of them have tests here).